packages/backend/src/core/reports/custom/report.ts

```diff
@@ -15,12 +15,15 @@ function createCustomTokenReport({
 }: {
   tokenPriceUsd: number
   tokenDecimals: number
-  tokenBalance: bigint
+  tokenBalance: bigint | ((t: UnixTime) => bigint)
   tokenId: AssetId
   projectId: ProjectId
   ethPriceUsd: number
   timestamp: UnixTime
 }): ReportRecord {
+  tokenBalance =
```

Comment:
I am not sure about this line, have you tested it? Maybe tokenBalance could always be a function?